src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -341,9 +307,22 @@ object JoinIndexRule
     require(resolve(spark, lRequiredIndexedCols, lRequiredAllCols).isDefined)
     require(resolve(spark, rRequiredIndexedCols, rRequiredAllCols).isDefined)
 
-    val lUsable = getUsableIndexes(lIndexes, lRequiredIndexedCols, lRequiredAllCols)
-    val rUsable = getUsableIndexes(rIndexes, rRequiredIndexedCols, rRequiredAllCols)
-    val compatibleIndexPairs = getCompatibleIndexPairs(lUsable, rUsable, lRMap)
+    val lUsable = getUsableIndexes(allIndexes, lRequiredIndexedCols, lRequiredAllCols)
+    val rUsable = getUsableIndexes(allIndexes, rRequiredIndexedCols, rRequiredAllCols)
+
+    // Get candidate via file-level metadata validation. This is performed after pruning
+    // by column schema, as this might be expensive when there are numerous files in the
+    // relation or many indexes to be checked.
+    val lIndexes = RuleUtils.getCandidateIndexes(
+      lUsable,
+      RuleUtils.getLogicalRelation(left).get,
```

Comment:
is it guaranteed that `getLogicalRelation` would always return `Some` at this point?

Reply:
Thanks! I added the condition in `isApplicable`